qcodes/instrument_drivers/tektronix/keithley_7510.py

```diff
@@ -1,10 +1,205 @@
-from typing import cast
+import numpy as np
+from typing import cast, Optional, List, Union
 
 from qcodes import VisaInstrument, InstrumentChannel
-from qcodes.utils.validators import Enum, Numbers, Ints
+from qcodes.instrument.parameter import invert_val_mapping
+from qcodes.utils.validators import Enum, Numbers, Ints, Lists
 from qcodes.utils.helpers import create_on_off_val_mapping
 
 
+class UnimplementedError(Exception):
+    pass
+
+
+class Buffer7510(InstrumentChannel):
+    """
+    Treat the reading buffer as a submodule, similar to Sense
+    """
+    default_buffer = {"defbuffer1", "defbuffer2"}
+
+    buffer_elements = {
+        "date": "DATE",
+        "measurement_formatted": "FORMatted",
+        "fractional_seconds": "FRACtional",
+        "measurement": "READing",
+        "relative_time": "RELative",
+        "seconds": "SEConds",
+        "source_value": "SOURce",
+        "source_value_formatted": "SOURFORMatted",
+        "source_value_status": "SOURSTATus",
+        "source_value_unit": "SOURUNIT",
+        "measurement_status": "STATus",
+        "time": "TIME",
+        "timestamp": "TSTamp",
+        "measurement_unit": "UNIT"
+    }
+
+    inverted_buffer_elements = invert_val_mapping(buffer_elements)
+
+    def __init__(
+            self,
+            parent: 'Keithley7510',
+            name: str,
+            size: Optional[int] = None,
+            style: str = ''
+    ) -> None:
+        super().__init__(parent, name)
+        self.buffer_name = name
```

Comment:
Why is `self.buffer_name` needed. It seems like this will always be equivalent to self.short_name which is already defined on all instruments

Reply:
It turns out `self.short_name` works. (not `self.name` though)

I was not sure if I should use `self.name` or `self.short_name`, so I created my own... :D